### PR TITLE
fix(neon): option serialization and reset

### PR DESCRIPTION
**Pull request by Leptopoda**
... getting there eventually.

Just tested it to a limited degree and I'd love to check everything to finally get it done right but review is still welcome.

### CODE REVIEW

File: packages/neon/neon/lib/src/settings/models/select_option.dart [hunk 1]
```diff
@@ -46,23 +44,27 @@ class SelectOption<T> extends Option<T> {
...
   Map<T, LabelBuilder> _values;
 
   @override
   set value(final T value) {
     super.value = value;
-    unawaited(storage.setString(key, serialize()));
+
+    if (value != null) {
+      // ignore: discarded_futures
+      storage.setString(key, serialize()!);
```
Comment (provokateurin) [inline packages/neon/neon/lib/src/settings/models/select_option.dart:66]: Better use unawaited

File: packages/neon/neon/lib/src/settings/models/toggle_option.dart [hunk 2]
```diff
@@ -25,15 +23,25 @@ class ToggleOption extends Option<bool> {
...
+    super.reset();
+  }
+
   @override
   set value(final bool value) {
     super.value = value;
-    unawaited(storage.setBool(key, serialize()));
+
+    // ignore: discarded_futures
+    storage.setBool(key, serialize());
```
Comment (provokateurin) [inline packages/neon/neon/lib/src/settings/models/toggle_option.dart:39]: unawaited here as well

**Review by provokateurin — APPROVED** [review]

File: packages/neon/neon/lib/src/settings/models/select_option.dart (hunk 1 above)
Comment (Leptopoda) [reply packages/neon/neon/lib/src/settings/models/select_option.dart:66]: Not possible as the type is FutureOr and unawaited needs a plain future.

I quite like using this in interfaces but I should just revert that :thinking:

Comment (Leptopoda) [reply packages/neon/neon/lib/src/settings/models/select_option.dart:66]: Not possible as the type is FutureOr and unawaited needs a plain future.

I quite like using this in interfaces but I should just revert that :thinking:

**Review by provokateurin — APPROVED** [review]

**Pull request merged**